code/datums/extensions/health/_health.dm

```diff
@@ -0,0 +1,175 @@
+/*
+ * General standardized health processing for all atoms.
+ * Current health and max health are always rounded to the nearest whole number.
+ */
+/datum/extension/health
+	base_type = /datum/extension/health
+	expected_type = /atom
+
+	/// Current health, from `0` to `max_health`.
+	var/current_health
+
+	/// Maximum health. `0` or `NULL` means no health handling.
+	var/max_health
+
+	/// Whether or not the death state has been set.
+	var/death_state = FALSE
+
+	/// LAZY List of damage type resistance or weakness multipliers, decimal form. Only applied to health reduction.
+	var/list/damage_resistances
+
+	/// Minimum damage required to actually affect health in `can_damage_health()`
+	var/min_damage_threshhold = 0
+
+
+/**
+ * Retrieves the extension's `current_health`.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/current_health()
+	return current_health
+
+
+/**
+ * Retrieves the extension's `max_health`.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/max_health(real = FALSE)
+	var/atom/A = holder
+	return real ? max_health : A.get_max_health()
+
```

Comment:
This file has a ton of unnecessary whitespace. Not necessarily a blocker but it looks untidy.

Reply:
Old habits of having two blank lines between functions/classes/etc. I can reduce it down to 1 line break per.